src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_ListViewBase.cs

```diff
@@ -2478,6 +2479,44 @@ static WeakReference AddItem(ObservableCollection<TestReleaseObject> collection)
 			}
 		}
 
+		[TestMethod]
+		public async Task When_Binding_and_Item_Removed()
+		{
+			const int ITEMS_TO_ADD = 6;
+			const int INDEX_TO_DELETE = 3;
+			using (FeatureConfigurationHelper.UseListViewAnimations())
+			{
+				var source = Enumerable.Range(0, ITEMS_TO_ADD).Select(a => new DefaultItem { Name = $"Item {a}", Value = a * a }).ToArray();
+
+				var SUT = new ListView
+				{
+					Width = 200,
+					Height = 300,
+					ItemTemplate = DefaultItemTemplate
+				};
+
+				var model = new When_Deleting_Item_DataContext(source);
+				SUT.DataContext = model;
+				SUT.SetBinding(ItemsControl.ItemsSourceProperty, new Binding { Path = new PropertyPath(nameof(model.Items)), Mode = BindingMode.OneWay });
+
+				WindowHelper.WindowContent = SUT;
+
+				await WindowHelper.WaitForLoaded(SUT);
+
+				Assert.AreEqual(ITEMS_TO_ADD, SUT.Items.Count);
+
+				var container = SUT.ContainerFromIndex(INDEX_TO_DELETE) as ContentControl;
+
+				model.Items.RemoveAt(INDEX_TO_DELETE);
+
+				// Ensure the container has properly been cleaned
+				// up after being removed.
+				Assert.IsNull(container.Content);
```

Comment:
Let's also assert that there's no expression on Content (just in case we try to add it back for some reason)

Reply:
@jeromelaban  I added the assertion, and it was failing. I had to add the fix we had initially discussed.